accord-core/src/main/java/accord/topology/Topology.java

```diff
@@ -323,6 +321,35 @@ else if (a[ai] < b[bi])
         }
     }
 
+    public <P1, P2, P3, O> O mapReduceOn(Id on, int offset, IndexedTriFunction<? super P1, ? super P2, ? super P3, ? extends O> function, P1 p1, P2 p2, P3 p3, BiFunction<? super O, ? super O, ? extends O> reduce, O initialValue)
+    {
+        NodeInfo info = nodeLookup.get(on);
+        if (info == null)
+            return initialValue;
+        int[] a = supersetIndexes, b = info.supersetIndexes;
```

Comment:
supersetIndexes seems to be the integer shard ids. For the entire topology it is just the sequence of numbers, but for `NodeInfo` is is the array of integers of shards it replicates.


Could the name change or add a comment?

Reply:
supersetIndexes is not the shard ids, it is the indexes in which the shards occur. For a `Topologies` that hasn't been sliced it will be an array containing the integers [0..size) for simplicity. I'm not sure what you're asking to change?

Reply:
Hmm so why create an array of integers instead of an array of object reference to the actual shards? It's twice the space sure, but simpler.

A comment saying that `Superset indexes are indexes of the shards that are present in this topology (or node info) out of the shards array` would make it quick to understand.

Reply:
Because operating on integers is much more efficient, particularly intersecting two different arrays. It’s nothing to do with storage efficiency.